streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -918,38 +630,96 @@ public synchronized Topology addStateStore(final StoreBuilder<?> storeBuilder,
+     * @param keyDeserializer
+     *        the {@link Deserializer} for record keys
+     *        (can be {@code null} to use the default key deserializer from {@link StreamsConfig})
+     * @param valueDeserializer
+     *        the {@link Deserializer} for record values
+     *        (can be {@code null} to use the default value deserializer from {@link StreamsConfig})
+     * @param topic
+     *        the source topic to read the data from
+     * @param processorName
+     *        the unique name of the internally added
+     *        {@link #addProcessor(String, ProcessorSupplier, String...) processor} which maintains the state store
+     * @param stateUpdateSupplier
+     *        the supplier used to obtain {@link Processor} instances, which maintain the state store
+     *
+     * @return itself
+     *
+     * @throws TopologyException
+     *         if the {@link StoreBuilder#name() state store} was already added, or
+     *         if the source or processor names are not unique, or
+     *         if the source topic has already been registered by another
+     *         {@link #addSink(String, String, String...) source}, read-only state store, or
+     *         {@link #addGlobalStore(StoreBuilder, String, Deserializer, Deserializer, String, String, ProcessorSupplier) global state store}
+     */
+    public synchronized <K, V> Topology addReadOnlyStateStore(final StoreBuilder<?> storeBuilder,
+                                                              final String sourceName,
+                                                              final Deserializer<K> keyDeserializer,
+                                                              final Deserializer<V> valueDeserializer,
+                                                              final String topic,
+                                                              final String processorName,
+                                                              final ProcessorSupplier<K, V, Void, Void> stateUpdateSupplier) {
+        return addReadOnlyStateStore(
+            storeBuilder,
+            sourceName,
+            null,
+            keyDeserializer,
+            valueDeserializer,
+            topic,
+            processorName,
+            stateUpdateSupplier
+        );
+    }
 
+    /**
+     * See {@link #addReadOnlyStateStore(StoreBuilder, String, Deserializer, Deserializer, String, String, ProcessorSupplier)}.
+     */
+    public synchronized <K, V> Topology addReadOnlyStateStore(final StoreBuilder<?> storeBuilder,
+                                                              final String sourceName,
+                                                              final TimestampExtractor timestampExtractor,
+                                                              final Deserializer<K> keyDeserializer,
+                                                              final Deserializer<V> valueDeserializer,
```

Comment:
Keeping generic types here does make sense, as it allows us to enforce the same type for the key and value on the `Deserializer` and the `ProcessorSupplier` (there is a few other places like this).

Given this one, and the `addProcessorCase`, and wondering if we should just add generic everywhere avoiding `<?>` just for code style consistency, even if it does not buy us anything for type safetly.

Thoughts?

Reply:
+1 for keeping the generic types over `<?>` for consistency